cmd/telegraf/agent.conf

```diff
@@ -94,3 +94,14 @@
   ## stateful plugins on termination of Telegraf. If the file exists on start,
   ## the state in the file will be restored for the plugins.
   # statefile = ""
+
+  ## If set to true, the agent will persist it's state at a regular interval set
+  ## by statefile_persist_interval.
+  # enable_statefile_persist_interval = false
+
+  ## The interval of time at which the state is peristed into the statefile.
+  # statefile_persist_interval = "10m"
```

Comment:
What if we combine both into one config option?

```suggestion
  ## The interval of time at which the state is peristed into the statefile.
  ## Default 0s means the agent will not persist the state at intervals.
  ## example: statefile_persist_interval = "10m"
  # statefile_persist_interval = "0s"
```

Reply:
Makes sense. The amount of options was getting a bit lengthy anyways.

Reply:
Although it could lead to a bit of confusion as the behavior for the other 'intervals' dont disable if set to zero.

Reply:
There is also no option to disable those, they are a core part of the working of telegraf.

Reply:
Should be resolved as of this [commit](https://github.com/influxdata/telegraf/pull/14561/commits/f835b958fb0c373c5e7e6751e45d50227ee59c7c)